packages/components/inputs/time-input/src/time-input.tsx

```diff
@@ -11,21 +10,120 @@ import { useFieldId, usePrevious } from '@commercetools-uikit/hooks';
 import TimeInputBody from './time-input-body';
 import messages from './messages';
 
+type ParsedTime = {
+  hours: number;
+  minutes: number;
+  seconds: number;
+  milliseconds: number;
+};
+
+export type TTimeInput = {
+  /**
+   * Used as HTML id property. An id is auto-generated when it is not specified.
+   */
+  id?: string;
+  /**
+   * Horizontal size limit of the input fields.
+   */
+  horizontalConstraint?:
+    | 1
+    | 2
+    | 3
+    | 4
+    | 5
+    | 6
+    | 7
+    | 8
+    | 9
+    | 10
+    | 11
+    | 12
+    | 13
+    | 14
+    | 15
+    | 16
+    | 'scale'
+    | 'auto';
+  /**
+   * Used as HTML name of the input component.
+   */
+  name?: string;
+  /**
+   * Used as HTML autocomplete of the input component.
+   */
+  autoComplete?: string;
+  /**
+   * Value of the input
+   */
+  value?: string;
+  /**
+   * Called with an event holding the new value.
+   * <br/>
+   * Required when input is not read only. Parent should pass it back as `value`-
+   * <br />
+   * Signature: `(event) => void`
+   */
+  onChange: (event: string | unknown) => void;
+  /**
+   * Called when input is blurred
+   * <br/>
+   * Signature: `(event) => void`
+   */
+  onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
+  /**
+   * Called when input is focused
+   * <br/>
+   * Signature: `(event) => void`
+   */
+  onFocus?: () => void;
```

Comment:
I think we can improve the event handlers a bit. 

```suggestion
type TEvent = {
  target: {id: string; name?: string; value: unknown}
}
  onChange: (event: TEvent) => void;
  /**
   * Called when input is blurred
   * <br/>
   * Signature: `(event) => void`
   */
  onBlur?: FocusEventHandler
  /**
   * Called when input is focused
   * <br/>
   * Signature: `(event) => void`
   */
  onFocus?: FocusEventHandler
```